pkg/commands/mcp/actions/mcp.go

```diff
@@ -0,0 +1,302 @@
+
+	requestBody, err := json.Marshal(request)
+	if err != nil {
+		utils.McpLogf("ERROR: Failed to marshal request: %v", err)
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to send the MCP request to Apono",
+				"data":    "Failed to marshal request",
+			},
+		}, EmptyErrorStatusCode
+	}
+
+	utils.McpLogf("Request body: %s", string(requestBody))
+
+	httpReq, err := http.NewRequestWithContext(context.Background(), "POST", endpoint, bytes.NewBuffer(requestBody))
+	if err != nil {
+		utils.McpLogf("ERROR: Failed to create HTTP request: %v", err)
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to send the MCP request to Apono",
+				"data":    "Failed to create HTTP request",
+			},
+		}, EmptyErrorStatusCode
+	}
+
+	httpReq.Header.Set("Content-Type", "application/json")
+
+	utils.McpLogf("Sending HTTP request...")
+	resp, err := httpClient.Do(httpReq)
+	if err != nil {
+		utils.McpLogf("ERROR: Failed to send HTTP request: %v", err)
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to send the MCP request to Apono",
+				"data":    fmt.Sprintf("Failed to send HTTP request: %v", err),
+			},
+		}, EmptyErrorStatusCode
+	}
+	defer func(Body io.ReadCloser) {
+		closeErr := Body.Close()
+		if closeErr != nil {
+			utils.McpLogf("WARNING: Failed to close response body: %v", closeErr)
+		}
+	}(resp.Body)
+
+	utils.McpLogf("HTTP response status: %d", resp.StatusCode)
+
+	if resp.StatusCode == http.StatusForbidden {
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    AuthError,
+				"message": "Authentication failed. Please run 'apono login' to authenticate.",
+				"data":    fmt.Sprintf("Status code %v - Invalid or expired authentication token", resp.StatusCode),
+			},
+		}, resp.StatusCode
+	}
+
+	// For notification methods, don't read response body - just log completion
+	if isNotificationsMethod(request.Method) {
+		utils.McpLogf("Notification method %s completed successfully with status %d", request.Method, resp.StatusCode)
+		return McpResponse{}, resp.StatusCode
+	}
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		utils.McpLogf("ERROR: Failed to read response body: %v", err)
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to send the MCP request to Apono",
+				"data":    "Failed to read response body",
+			},
+		}, resp.StatusCode
```

Comment:
you have the same block of code creating almost the same internal error over and over, wrap into a methof